File: 2. Задачи с ЕГЭ по информатике/E. Дед Мороз и Снегурочка.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
	int N, D, K, R;
	cin >> N;
	int c[100];
	for (int i = 0; i < 100; i++) 
		c[i] = 0;
	for (int i = 0; i < N; i++) {
		cin >> D >> K;
		R = K % D;
		if (R > 0) c[R]++;
	}
	int imax = 1;
	for (int i = 0; i < 100; i++) {
		if (c[i] >= c[imax])
			imax = i;
	}
	if (c[imax] == 0)
		imax = 0;
	cout << imax;
	return 0;
}

File: 2. Задачи с ЕГЭ по информатике/E. Дед Мороз и Снегурочка.py
DMAX = 100
c = [0 for i in range(DMAX)]
N = int(input())
for i in range(N):
	D, K = map(int, input().split())
	r = K % D
	if r > 0:
		c[r] += 1
imax = 1
for i in range(DMAX):
	if c[i] >= c[imax]:
		imax = i
if c[imax] == 0:
	imax = 0
print(imax)